File: src/main/java/xyz/ivyxjc/pubg4j/entity/PubgMatchDetail.java
package xyz.ivyxjc.pubg4j.entity;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;
import lombok.AccessLevel;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import xyz.ivyxjc.pubg4j.types.GameMode;
import xyz.ivyxjc.pubg4j.types.MapName;
import xyz.ivyxjc.pubg4j.types.PlatformRegion;

/**
 * @author Ivyxjc
 * @since 4/22/2018
 */
@Data
public class PubgMatchDetail {

    private String matchId;
    /**
     * Time this match object was stored in the API
     */
    private LocalDateTime createdAt;

    /**
     * Length of the match
     */
    private Integer duration;

    /**
     * Game mode played
     */
    private GameMode gameMode;

    /**
     * Map name
     */
    private MapName mapName;

    /**
     * Platform-region shard
     */
    private PlatformRegion shardId;
    /**
     * Identifies the studio and game
     */
    private String titleId;

    private Map<String, PubgRoster> rosterMap;

    private Map<String, PubgParticipant> participantDetailMap;

    @Getter(AccessLevel.NONE)
    @Setter(AccessLevel.NONE)
    private String patchVersion;

    @Getter(AccessLevel.NONE)
    @Setter(AccessLevel.NONE)
    private String stats;

    @Getter(AccessLevel.NONE)
    @Setter(AccessLevel.NONE)
    private String tags;

    public PubgMatchDetail() {
        rosterMap = new HashMap<>();
        participantDetailMap = new HashMap<>();
    }
}

File: src/main/kotlin/xyz/ivyxjc/pubg4j/entity/PubgMatchDetail.kt
package xyz.ivyxjc.pubg4j.entity

import xyz.ivyxjc.pubg4j.types.GameMode
import xyz.ivyxjc.pubg4j.types.MapName
import xyz.ivyxjc.pubg4j.types.PlatformRegion
import java.time.LocalDateTime
import java.util.*

/**
 * @author Ivyxjc
 * @since 4/22/2018
 */
class PubgMatchDetail {

    constructor()

    constructor(matchId: String) {
        this.matchId = matchId
    }

    lateinit var matchId: String
    /**
     * Time this match object was stored in the API
     */
    var createdAt: LocalDateTime? = null

    /**
     * Length of the match
     */
    var duration: Int? = null

    /**
     * Game mode played
     */
    var gameMode: GameMode? = null

    /**
     * Map name
     */
    var mapName: MapName? = null

    /**
     * Platform-region shard
     */
    var shardId: PlatformRegion? = null

    /**
     * Identifies the studio and game
     */
    var titleId: String? = null

    val rosterMap: MutableMap<String, PubgRoster>

    val participantDetailMap: MutableMap<String, PubgParticipant>

    private val patchVersion: String? = null

    private val stats: String? = null

    private val tags: String? = null

    init {
        rosterMap = HashMap()
        participantDetailMap = HashMap()
    }
}
